sydent/db/valsession.py

```diff
@@ -81,11 +86,25 @@ def setSendAttemptNumber(self, sid, attemptNo):
         cur.execute("update threepid_token_auths set sendAttemptNumber = ? where id = ?", (attemptNo, sid))
         self.sydent.db.commit()
 
-    def setValidated(self, sid, validated):
+    def setValidated(self, sid, validated, commit=True):
+        """Set the validation status of a threepid validation session
+
+        :param sid: The session ID
+        :type sid: str
+
+        :param validated: Whether a session is validated or not
+        :type validated: bool
+
+        :param commit: Whether to commit to the database. Used in applying other
+            database operations atomically. Defaults to True
+        :type bool: bool
+        """
         cur = self.sydent.db.cursor()
 
         cur.execute("update threepid_validation_sessions set validated = ? where id = ?", (validated, sid))
-        self.sydent.db.commit()
+
+        if commit:
+            self.sydent.db.commit()
```

Comment:
I'm not convinced this works if you create a new cursor? Given this is a trivial function I'd be tempted to move `cur.execute(..)` inline

Reply:
> I'm not convinced this works if you create a new cursor? 

```
>>> cur = con.cursor()
>>> cur.execute("insert into bla (something) values (1)")
<sqlite3.Cursor object at 0x7f1cf9c65960>
>>> cur = con.cursor()
>>> cur.execute("insert into bla (something) values (2)")
<sqlite3.Cursor object at 0x7f1cf9c65880>
>>> con.commit()
>>> cur = con.cursor()
>>> rows = cur.execute("select * from bla")
>>> rows.fetchone()
(1,)
>>> rows.fetchone()
(2,)
>>> rows.fetchone()
>>> 
```

seems to work.

By inline you mean just drop the function and call `execute` from `validateSessionWithToken` directly?

Reply:
> seems to work

"Maybe", though god only know what cursor actually is doing under the hood. Randomly using new cursors doesn't seem necessary like the way to go

> By inline you mean just drop the function and call `execute` from `validateSessionWithToken` directly?

yup

Reply:
How's it looking now?